db.go

```diff
@@ -238,7 +238,7 @@ func (db *DB) mmap(minsz int) error {
 	}
 
 	// Memory-map the data file as a byte slice.
-	if err := mmap(db, size); err != nil {
+	if err := mmap(db, size, int(info.Size()) != size); err != nil {
```

Comment:
Can you type convert `size` up to an `int64` instead of `info.Size()` down to an `int`?


e.g.

```
info.Size() != int64(size)
```

Theoretically a 32-bit machine could have a 4GB upper limit whereas `int` on a 32-bit machine has an upper limit of ~2B.